cli/azd/pkg/project/service_target_ml_endpoint.go

```diff
@@ -0,0 +1,280 @@
+package project
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/machinelearning/armmachinelearning/v3"
+	"github.com/azure/azure-dev/cli/azd/pkg/ai"
+	"github.com/azure/azure-dev/cli/azd/pkg/async"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+)
+
+// machineLearningEndpointTarget is a ServiceTarget implementation for deploying to Azure ML online endpoints
+type machineLearningEndpointTarget struct {
+	env        *environment.Environment
+	envManager environment.Manager
+	aiHelper   AiHelper
+}
+
+// NewMachineLearningEndpointTarget creates a new machineLearningEndpointTarget instance
+func NewMachineLearningEndpointTarget(
+	env *environment.Environment,
+	envManager environment.Manager,
+	aiHelper AiHelper,
+) ServiceTarget {
+	return &machineLearningEndpointTarget{
+		env:        env,
+		envManager: envManager,
+		aiHelper:   aiHelper,
+	}
+}
+
+// EndpointDeploymentResult is a struct to hold the result of an online endpoint deployment
+type EndpointDeploymentResult struct {
+	Environment *armmachinelearning.EnvironmentVersion
+	Model       *armmachinelearning.ModelVersion
+	Flow        *ai.Flow
+	Deployment  *armmachinelearning.OnlineDeployment
+}
+
+// Initialize initializes the machineLearningEndpointTarget
+func (m *machineLearningEndpointTarget) Initialize(ctx context.Context, serviceConfig *ServiceConfig) error {
+	return serviceConfig.Project.AddHandler(
+		"postprovision",
```

Comment:
Does it happen to be used only when running azd up (and having provision + deploy) ?

Trying to understand why/how a service target is knowing here something about `postprovision`

shouldn't this be happening as part of `package` , maybe?

Reply:
The goal was to support exposing additional links that are displayed to the user at the end of an `azd provision`.  @ellismg offered a good recommendation to enable a new method on the service targets for exposing links that we can display in various stages of the CLI.  

This currently happens anytime after any provision event fires within the project (anytime `azd provision` is invoked.)

Reply:
Rolling this bit back for now till we have a better feature to support it.